test/cpp/ext/gcp/environment_autodetect_test.cc

```diff
@@ -55,25 +54,23 @@ class EnvironmentAutoDetectTest : public ::testing::Test {
 TEST_F(EnvironmentAutoDetectTest, Basic) {
   grpc::internal::EnvironmentAutoDetect env("project");
 
-  absl::Notification notify;
+  grpc_core::Notification notify;
   GetNotifiedOnEnvironmentDetection(&env, &notify);
   notify.WaitForNotification();
 
   // Unless we test in a specific GCP resource, we should get "global" here.
-  EXPECT_EQ(env.resource()->resource_type, "global");
-  EXPECT_THAT(env.resource()->labels,
-              UnorderedElementsAre(Pair("project_id", "project")));
+  // EXPECT_EQ(env.resource()->resource_type, "global");
```

Comment:
Why is this commented out?

Reply:
This is a problem of our testing environments. I can't be sure whether we'd be running on an environment that has MetadataServer or not, so unless I mock the MetadataServer, I can not get a stable result here across environments.